satpy/composites/__init__.py

```diff
@@ -1039,15 +1059,28 @@ def __init__(self, name, filename=None, url=None, known_hash=None, area=None,
 
     @staticmethod
     def _get_cache_filename_and_url(filename, url):
-        if filename is not None:
+        def _check_relative_filename(filename):
+            data_dir = satpy.config.get('data_dir')
+            path = os.path.join(data_dir, filename)
+
+            return path if os.path.exists(path) else filename
```

Comment:
Could you make this a separate method on the class? If you have to make the calling method not static that's fine. You could also do `StaticImageCompositor._check_relative_filename` but maybe that's too ugly.

Reply:
Sure, if you are fine with that I change the calling method to non static, I can move it.